src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1047,4 +1053,53 @@ private LegacySQLTypeName translateType(String datatype) {
             default: throw new IllegalArgumentException("Unknown datatype '" + datatype + "'");
         }
     }
+
+    private String externalTableName(String tableName, String suffix) {
+        return String.format("%s%s_%s", PDAO_EXTERNAL_TABLE_PREFIX, tableName, suffix);
+    }
+
+    public String createExternalTable(Dataset dataset, String path, String tableName, String suffix) {
```

Comment:
From the flight code I see that both `path` and `tableName` are derived from the same `table` object. Could we refactor this to pass that `table` in here? It'd make the flight code simpler, and reduce the chances of somebody passing in mismatched args.

Reply:
Also, on naming: This is named really generically, but hard-coded to a specific use-case. Should we either:
* Rename the method to reflect its specific use, or
* Add more parameters to the method for the `Schema` and `FormatOptions`, and let the flight be the one to do the hard-coding for now?

Reply:
Good point, will go with the second option.

Reply:
Changed my mind and went with first. I prefer to avoid "leaking" BQ objects out into the steps.. it seems like we get better encapsulation having the PDAO be responsible for handling things like organizing names and dealing with `Schema` creation, etc.